libsolidity/AST.h

```diff
@@ -315,6 +334,31 @@ class StructDefinition: public Declaration
 	std::vector<ASTPointer<VariableDeclaration>> m_members;
 };
 
+class EnumDefinition: public Declaration
+{
+public:
+	EnumDefinition(Location const& _location,
+				   ASTPointer<ASTString> const& _name,
+				   std::vector<ASTPointer<EnumValue>> const& _members):
+		Declaration(_location, _name), m_members(_members) {}
+	virtual void accept(ASTVisitor& _visitor) override;
+	virtual void accept(ASTConstVisitor& _visitor) const override;
+
+	std::vector<ASTPointer<EnumValue>> const& getMembers() const { return m_members; }
+
+	virtual TypePointer getType(ContractDefinition const*) const override;
+
+	/// @returns the value that the string has in the Enum
+	unsigned int getMemberValue(ASTString const& _member) const;
```

Comment:
This is non-syntactic information, can you move that to the type?
Perhaps such a function should even be available in the class `MemberList` - as we noticed that the list of members should really be a list, it makes sense to assign indices to the members.